crypto/asn1/asn1_lib.c

```diff
@@ -272,13 +272,20 @@ int ASN1_STRING_set(ASN1_STRING *str, const void *_data, int len)
 {
     unsigned char *c;
     const char *data = _data;
+    size_t new_len = 0;
 
     if (len < 0) {
         if (data == NULL)
             return 0;
         else
-            len = strlen(data);
+            new_len = strlen(data);
     }
+
+    if ( new_len > INT_MAX)
```

Comment:
style: `if (new_len ...`

Reply:
Please also push an error to the stack before returning:

```patch
diff --git a/crypto/asn1/asn1_lib.c b/crypto/asn1/asn1_lib.c
index 488aa92744..bed3862628 100644
--- a/crypto/asn1/asn1_lib.c
+++ b/crypto/asn1/asn1_lib.c
@@ -281,10 +281,13 @@ int ASN1_STRING_set(ASN1_STRING *str, const void *_data, int len)
             new_len = strlen(data);
     }
 
-    if ( new_len > INT_MAX)
+    if (new_len > INT_MAX) {
+        ASN1err(ASN1_F_ASN1_STRING_SET, ASN1_R_LENGTH_OUT_OF_RANGE);
         return 0;
-    else
+    }
+    else {
         len = (int)new_len;
+    }
 
     if ((str->length <= len) || (str->data == NULL)) {
         c = str->data;
```
(Note: The constant `ASN1_R_LENGTH_OUT_OF_RANGE` does not exist yet. After editing the file, you have to run `make update`, this will generate the necessary definitions, which you then add to the same commit.)